src/components/faceVerification/components/ErrorButtons.js

```diff
@@ -1,20 +1,49 @@
-import React from 'react'
+import React, { useCallback } from 'react'
 import { View } from 'react-native'
+import { t } from '@lingui/macro'
 
+import Config from '../../../config/config'
 import GiveUpButton from '../standalone/components/GiveUpButton'
 import { CustomButton } from '../../common'
 import useVerificationAttempts from '../hooks/useVerificationAttempts'
+import { openLink } from '../../../lib/utils/linking'
+import { getDesignRelativeHeight } from '../../../lib/utils/sizes'
+import { withStyles } from '../../../lib/styles'
 
 const ErrorButtons = ({ styles, screenProps, navigation, onRetry }) => {
   const { isReachedMaxAttempts } = useVerificationAttempts()
```

Comment:
this flag resets when read. you should read it strictly once in ErrorScreen then just pass down to the nested components

```javascript
import React, { useCallback, useContext, useEffect, useMemo, useState } from 'react'
import { get } from 'lodash'
import { View } from 'react-native'

import CameraNotAllowedError from '../components/CameraNotAllowedError'
import DeviceOrientationError from '../components/DeviceOrientationError'
import DuplicateFoundError from '../components/DuplicateFoundError'
import NotMatchError from '../components/NotMatchError'
import GeneralError from '../components/GeneralError'
import UnrecoverableError from '../components/UnrecoverableError'
import SwitchToAnotherDevice from '../components/SwitchToAnotherDevice'

import useVerificationAttempts from '../hooks/useVerificationAttempts'

import { getFirstWord } from '../../../lib/utils/getFirstWord'
import useProfile from '../../../lib/userStorage/useProfile'
import { FVFlowContext } from '../standalone/context/FVFlowContext'
import { Wrapper } from '../../common'
import ErrorButtons from '../components/ErrorButtons'
import { withStyles } from '../../../lib/styles'
import { getDesignRelativeHeight, getDesignRelativeWidth } from '../../../lib/utils/sizes'

const getStylesFromProps = ({ theme }) => ({
  topContainer: {
    alignItems: 'center',
    justifyContent: 'space-evenly',
    display: 'flex',
    backgroundColor: theme.colors.surface,
    height: '100%',
    flex: 1,
    flexGrow: 1,
    flexShrink: 0,
    paddingBottom: getDesignRelativeHeight(theme.sizes.defaultDouble),
    paddingLeft: getDesignRelativeWidth(theme.sizes.default),
    paddingRight: getDesignRelativeWidth(theme.sizes.default),
    paddingTop: getDesignRelativeHeight(theme.sizes.defaultDouble),
    borderRadius: 5,
    marginBottom: theme.paddings.bottomPadding,
  },
})

const ErrorScreen = ({ styles, screenProps, navigation }) => {
  const profile = useProfile()
  const { isReachedMaxAttempts } = useVerificationAttempts()
  const { isFVFlow } = useContext(FVFlowContext)

  const exception = get(screenProps, 'screenState.error')
  const kindOfTheIssue = get(exception, 'name')

  const title = useMemo(() => {
    const { fullName } = profile

    return getFirstWord(fullName)
  }, [profile])

  const onRetry = useCallback(() => screenProps.navigateTo('FaceVerificationIntro'), [screenProps])

  const [{ ErrorViewComponent, reachedMax }, setDisplayState] = useState({})

  useEffectOnce(() => {
    // determining error component to display
    // be default display general error
    let component = GeneralError
    const { kindOfTheIssue: map } = ErrorScreen
    const reachedMax = isReachedMaxAttempts()

    // if reached max retries - showing 'something went wrong our side'
    if (reachedMax && 'UnrecoverableError' !== kindOfTheIssue) {
      component = SwitchToAnotherDevice

      // otherwise, if there's special screen for this kind of the issue hapened - showing it
    } else if (kindOfTheIssue in map) {
      component = map[kindOfTheIssue]
    }

    setDisplayState({ ErrorViewComponent: component, reachedMax })
  }, [isReachedMaxAttempts, kindOfTheIssue, setDisplayState])

  if (!ErrorViewComponent) {
    return null
  }

  return (
    <Wrapper>
      <View style={styles.topContainer}>
        <ErrorViewComponent
          displayTitle={title}
          nav={screenProps}
          exception={exception}
          isFVFlow={isFVFlow}
          reachedMax={reachedMax}
        />
        <ErrorButtons onRetry={onRetry} navigation={navigation} reachedMax={reachedMax} />
      </View>
    </Wrapper>
  )
}

ErrorScreen.kindOfTheIssue = {
  NotMatchError,
  UnrecoverableError,
  DuplicateFoundError,
  DeviceOrientationError,
  NotAllowedError: CameraNotAllowedError,
  NotSupportedError: SwitchToAnotherDevice,
}

export default withStyles(getStylesFromProps)(ErrorScreen)
```

this one to lib/hooks if not exists yet:


```javascript

import { useRef, EffectCallback, useEffect } from "react"

export const useEffectOnce: typeof useEffect = (effect: EffectCallback) => {
  const initializingRef = useRef(false)

  useEffect(() => {
    if (initializingRef.current) {
      return
    }

    initializingRef.current = true
    return effect()
  }, [effect])
}

```

Reply:
Ah thanks, didnt know

fixed now